src/components/structures/RoomView.js

```diff
@@ -1199,9 +1279,22 @@ module.exports = React.createClass({
 
     sendReadReceipt: function() {
         if (!this.state.room) return;
+        if (!this.refs.messagePanel) return;
+
         var currentReadUpToEventId = this.state.room.getEventReadUpTo(MatrixClientPeg.get().credentials.userId);
         var currentReadUpToEventIndex = this._indexForEventId(currentReadUpToEventId);
 
+        // We want to avoid sending out read receipts when we are looking at
+        // events in the past.
+        //
+        // For now, let's apply a heuristic: if (a) the server has a
+        // readUpToEvent for us, (b) we can't find it, and (c) we could
+        // forward-paginate the event timeline, then suppress read receipts.
+        if (currentReadUpToEventId && currentReadUpToEventIndex === null &&
```

Comment:
yeah, the precise wording of the comment is wrong. But there shouldn't be any local-echoes, because we aren't sending any RRs, so the code reduces to the same thing. I'll update the comment.